app/src/main/java/com/example/musicdemomvp/ui/MainActivity.kt

```diff
@@ -0,0 +1,224 @@
+import com.example.musicdemomvp.presenter.SongPresenter
+import com.example.musicdemomvp.service.SongService
+import com.example.musicdemomvp.utils.Repository
+import com.example.musicdemomvp.utils.SongContract
+import java.text.SimpleDateFormat
+import java.util.*
+
+class MainActivity : AppCompatActivity(), SongContract.View, SeekBar.OnSeekBarChangeListener {
+
+    private var songPresenter: SongContract.Presenter? = null
+    private var songAdapter = SongAdapter(::clickSong)
+    private var songService: SongService? = null
+    private var musicBound = false
+    private var playIntent: Intent? = null
+    private var songList = mutableListOf<Song>()
+    private val permission = arrayOf(Manifest.permission.READ_EXTERNAL_STORAGE, Manifest.permission.WRITE_EXTERNAL_STORAGE)
+    private val musicConnection: ServiceConnection = object : ServiceConnection {
+        override fun onServiceConnected(name: ComponentName, service: IBinder) {
+            val binder = service as SongService.SongBinder
+            songService = binder.getService()
+            songService?.setList(songList as ArrayList<Song>)
+            musicBound = true
+        }
+
+        override fun onServiceDisconnected(name: ComponentName) {
+            musicBound = false
+        }
+    }
+    lateinit var binding: ActivityMainBinding
+    @SuppressLint("SimpleDateFormat")
+    val timeFormat = SimpleDateFormat("mm:ss")
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = DataBindingUtil.setContentView(this, R.layout.activity_main)
+
+        registerReceiver(notificationReceiver, IntentFilter(getString(R.string.notif_intent)))
+        songPresenter = SongPresenter(this, Repository.getInstance(SongLocalDataSource.getInstance(baseContext)))
+
+        if (checkPermission()) {
+            init()
+            binding.buttonNext.setOnClickListener { nextAction() }
+            binding.buttonPlay.setOnClickListener { playAction() }
+            binding.buttonPrevious.setOnClickListener { prevAction() }
+            binding.seekbarTime.setOnSeekBarChangeListener(this)
+        } else {
+            askForPermission()
+        }
+    }
+
+    override fun getSongSuccess(songs: List<Song>) {
+        songList = songs as MutableList<Song>
+        songAdapter.updateDataAdapter(songs)
+    }
+
+    override fun setPauseButton() = binding.buttonPlay.setImageResource(R.drawable.ic_pause)
+
+    override fun setPlayButton() = binding.buttonPlay.setImageResource(R.drawable.ic_play)
+
+    override fun resetSeekBar() {
+        binding.seekbarTime.progress = 0
+        binding.textTimeMin.text = getString(R.string.time_start)
+    }
+
+    override fun initPlay(index: Int) {
+        setPauseButton()
+        binding.textSongName.text = songList[index].title
+        binding.textTimeMax.text = timeFormat.format(songService?.getDuration())
+        binding.seekbarTime.max = songService?.getDuration()!!
```

Comment:
```suggestion
       songService?.getDuration()?.let { binding.seekbarTime.max = it } 
```
hạn chế dùng !! nhé, nó throw ra lỗi mà k xử lý là crash app đấy